spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -225,12 +229,29 @@ protected void handleMessage(Message<?> message) {
 
 	@Override
 	protected Object getResourceToBind() {
-		return this.source;
+		return this.originalSource;
 	}
 
 	@Override
 	protected String getResourceKey() {
 		return IntegrationResourceHolder.MESSAGE_SOURCE;
 	}
 
+	private static Object extractProxyTarget(Object target) {
+		if (!(target instanceof Advised)) {
+			return target;
+		}
+		Advised advised = (Advised) target;
+		if (advised.getTargetSource() == null) {
+			return null;
```

Comment:
Well, what would be better for us?
Looks like `AdvisedSupport` never returns null:
```
public void setTargetSource(TargetSource targetSource) {
	this.targetSource = (targetSource != null ? targetSource : EMPTY_TARGET_SOURCE);
}
```
But at the same time `EmptyTargetSource` returns `null` for the target.

Seems for me we can have a `null` in case we proxy just only interface, like we do in the `GatewayProxyFactoryBean`. So, to be safe I see only the way to fallback to the provided `source` instead of `null`.

WDYT?